Update documentation on computer setup #3014 (#3044)
aiidateam/aiida_core: docs/source/get_started/computers.rst

   .. tip:: You can press ``<CTRL>+C`` at any moment to abort the setup process.
     Nothing will be stored in the DB.
   
   
   Here is a list of what is asked, together with an explanation.
   

>    * **Computer label**: the (user-friendly) name of the new computer instance 

Reviewer: Could you please replace "name" by "label" in the text below as well?
Just to avoid confusion between name / label

        aprun -n {tot_num_mpiprocs}
        poe

   * **Default number of CPUs per machine**: The number of MPI processes per machine that
     should be executed if it is not otherwise specified. Use ``0`` to specify no default value. 
   
   At the end, the command will open your default editor on a file containing a summary
   of the configuration up to this point, and the possibility to add ``bash``
   commands that will be executed either *before* the actual execution of the job
   (under 'pre-execution script') or *after* the script submission (under 'Post execution script').
   These additional lines need may set up the environment on the computer,
   for example loading modules or exporting environment variables, for example::

        export NEWVAR=1
        source some/file

   .. note:: Don't specify settings here that are specific to a code, calculation or scheduler -- 
      you can set further pre-execution commands at the ``Code`` and ``CalcJob`` level.

   When you are done editing, save and quit (e.g. ``<ESC>:wq<ENTER>`` in ``vim``).
   The computer has now been created in the database but you still need to *configure* access to it
   using your credentials.

>    For the user convenience, it is also possible provide some (or all) the information

Reviewer: ```suggestion
   In order to avoid having to retype the setup information the next time round, it is also possible provide some (or all) of the information
```

>    described above from a configuration file, using the command::

Reviewer: ```suggestion
   described above via a configuration file using::
```

        verdi computer setup --config computer.yml

   where ``computer.yml`` is a configuration file in the
   `YAML format <https://en.wikipedia.org/wiki/YAML#Syntax>`_.

>    This file contains the information in a series of key:value entries, like it is shown below:

Reviewer: ```suggestion
   This file contains the information in a series of key:value pairs:
```

   .. code-block:: yaml

      ---
      label: "localhost"
      hostname: "localhost"
      transport: local
      scheduler: "direct"
      work_dir: "/home/max/.aiida_run"
      mpirun_command: "mpirun -np {tot_num_mpiprocs}"
      mpiprocs_per_machine: "2"
      prepend_text: |
        module load mymodule
        export NEWVAR=1

>   .. tip:: The list of the keys that can be used is available from the options flags of the command: ::

Reviewer: ```suggestion
  .. tip:: The keys mirror the available options of the command, which you can print using: ::
```

        verdi computer setup --help

>      but be aware of the syntax differences, namely the lack of initial ``--``,

Reviewer: ```suggestion
     Note the syntax differences: remove the ``--`` prefix
```

>      and any ``-`` within the keyare  substituted by the underscore ``_``.

Reviewer: ```suggestion
     and replace ``-`` within the keys by the underscore ``_``.
```

        aprun -n {tot_num_mpiprocs}
        poe

  * **Default number of CPUs per machine**: The number of MPI processes per machine that

>     should be executed if it is not otherwise specified. No default value is specified by 0. 

Reviewer: ```suggestion
    should be executed if it is not otherwise specified. Use ``0`` to specify no default value. 
```

>   At the end, you will land on an vim editor page, with a summary of the configuration up to this point, 

Reviewer: ```suggestion
  At the end, the command will open your default editor on a file containing a summary of the configuration up to this point, 
```

>   and the possibility to add content that will be executed either

Reviewer: ```suggestion
  and the possibility to add ``bash`` commands that will be executed either
```

>   *before* the actual execution of the job (under the 'pre-execution script' lines) or *after* the 

Reviewer: ```suggestion
  *before* the actual execution of the job (under 'pre-execution script') or *after* the 
```

>   script submission (under the 'Post execution script' lines). 

Reviewer: ```suggestion
  script submission (under 'Post execution script'). 
```

>   These additional lines need to be proper ``bash`` code, that is intended for computer-dependent code

Reviewer: ```suggestion
  These additional lines need may set up the environment on the computer, for example loading modules or exporting environment variables::```

>   (for instance a module that should always be loaded). For example::

Reviewer: ```suggestion

```

        export NEWVAR=1
        source some/file

>   .. note:: Pre-execution commands should **not** be used to define resources that are handled by the scheduler plugin, 

Reviewer: ```suggestion
  .. note:: Don't specify settings here that are specific to a code or calculation -- you can set further pre-execution commands at the ``Code`` and ``CalcJob`` level.
```

Reviewer 2: I think that Daniele meant that one should not add e.g. `$SBATCH -n 2` instructions. But you are also right.

Author: Yes, @borellim, exactly. I guess both can be left there @ltalirz ?

     like the number of nodes or execution time.

>   After finishing to edit the script, save and quit by typing ``<ESC>:wq<ENTER>``. The computer is created and needs

Reviewer: ```suggestion
  When you are done editing, save and quit (e.g. ``<ESC>:wq<ENTER>`` in ``vim``). The computer has now been created in the database but you still need to *configure* access to it using your credentials.
```
